frankDoc/src/main/java/nl/nn/adapterframework/frankdoc/doclet/FrankEnumConstant.java

```diff
@@ -0,0 +1,22 @@
+/* 
+Copyright 2021 WeAreFrank! 
+
+Licensed under the Apache License, Version 2.0 (the "License"); 
+you may not use this file except in compliance with the License. 
+You may obtain a copy of the License at 
+
+    http://www.apache.org/licenses/LICENSE-2.0 
+
+Unless required by applicable law or agreed to in writing, software 
+distributed under the License is distributed on an "AS IS" BASIS, 
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+See the License for the specific language governing permissions and 
+limitations under the License. 
+*/
+
+package nl.nn.adapterframework.frankdoc.doclet;
+
+public interface FrankEnumConstant extends FrankProgramElement {
```

Comment:
Hoort FrankEnumConstant niet in package model i.p.v. doclet? Geldt dat niet ook voor een aantal andere classes, zoals FrankProgramElement?

Reply:
Ik denk dat het zo goed is. Package ...doclet was ooit bedoeld om java reflectie te hebben die hetzelfde gebruikt kan worden binnen een doclet en runtime. Nu is de runtime implementatie weg. Dit package dus een wrapper van de doclet API.

Enum constanten worden door Java reflectie en door de doclet API beschouwd als fields. Ik gebruik package ...doclet om dit detail te verbergen voor het model. Package ...doclet bepaalt wat de enum constanten van een class zijn en verzamelt van die enum constanten de JavaDoc en de annotaties. Het is aan het model om die informatie te interpreteren. Bij het vullen van het model wordt besloten dat de JavaDoc comment van een enum constante de description is, en dat de @EnumLabel annotatie het label bevat. Het model heeft classes AttributeValues en AttributeValue. Daar zitten methods in om de enum waardes van een attribute te krijgen, en om het label en de description te geven.

Zie boven over de namen AttributeValues en AttributeValue; die namen zijn inderdaad niet goed.